language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ObjectBodiedNodeContextProvider.java

```diff
@@ -92,4 +97,16 @@ private boolean isClientObject(Node node) {
                 && ((ObjectConstructorExpressionNode) node).objectTypeQualifiers().stream()
                 .anyMatch(token -> token.kind() == SyntaxKind.CLIENT_KEYWORD));
     }
+
+    public static boolean onSuggestInitFunction(Node node) {
+        if (node.kind() != SyntaxKind.SERVICE_DECLARATION) {
+            return false;
+        }
+
+        ServiceDeclarationNode serviceDeclarationNode = (ServiceDeclarationNode) node;
+        return serviceDeclarationNode.members().stream()
+                .filter(member-> member.kind() == SyntaxKind.OBJECT_METHOD_DEFINITION)
+                .map(member->(FunctionDefinitionNode) member)
+                .noneMatch(funcDef -> "init".equals(funcDef.functionName().text()));
+    }
```

Comment:
You can all there together in to a single statement

Reply:
Also, move the `init` string to a constant/ check whether we have already defined it.